.github/workflows/main.yml

```diff
@@ -25,6 +27,24 @@ jobs:
         export VITE_BASE_DIR="/$(basename $GITHUB_REPOSITORY)/"
         echo "VITE_BASE_DIR: $VITE_BASE_DIR"
         ./run.sh
+    - name: Predeployment test
+      run: bash ./test.sh
+    - name: Upload Report Artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: reposense-report
+        path: ./reposense-report
+
+  Deploy:
+    needs: Test
+    runs-on: ubuntu-22.04
+    if: github.event_name == 'push'
```

Comment:
this should also defensively check (and for clarity when reading the workflow) that the pushed branch is master